src/shared/src/units.ts

```diff
@@ -0,0 +1,73 @@
+import { NXDataStore } from "./persistence";
+
+// SI base units
+export type Celsius = number; // derived unit
+export type HectoPascal = number; // derived unit
+export type KiloGram = number;
+export type Metre = number;
+
+export class Units {
+    private static _metricUnits: boolean;
+
+    static get metricUnits(): boolean {
+        if (Units._metricUnits === undefined) {
+            NXDataStore.getAndSubscribe<string>('CONFIG_USING_METRIC_UNIT', (key: string, value: string) => {
+                Units._metricUnits = value === '1';
+            });
+        }
+        return Units._metricUnits;
+    }
+
+    static userToKilogram(value: number): KiloGram {
+        return Units.metricUnits ? value : value / 2.20462;
+    }
+
+    static kilogramToUser(value: KiloGram): number {
+        return Units.metricUnits ? value : value * 2.20462;
+    }
+
+    static get userWeightUnit(): 'kg' | 'lbs' {
```

Comment:
**thought:** I'm not sure if the prefix should be the responsibility of this class.

If we don't want to refactor this a different way I would name this method `userWeightSuffixEis2` or something.

Reply:
How come you would prefer for this to not be within the scope of this class? I think it makes sense to be able to access prefixes and conversions from a 'Unit' class instead of splitting prefixes off into a 'Prefix' class